internal/app/starter/master_linux_test.go

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2018-2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package starter
+
+import (
+	"testing"
+
+	"github.com/sylabs/singularity/internal/pkg/runtime/engines"
+	"github.com/sylabs/singularity/internal/pkg/test"
+)
+
+func TestDoRPCThread(t *testing.T) {
+	test.DropPrivilege(t)
+	defer test.ResetPrivilege(t)
+
+	var fatal error
+	fatalChan := make(chan error, 1)
+
+	var fakeEngine engines.Engine
+
+	tests := []struct {
+		name         string
+		rpcSocket    int
+		containerPid int
+		engine       *engines.Engine
+		shallPass    bool
+	}{
+		{
+			name:         "invalid case",
+			rpcSocket:    -1,
+			containerPid: -1,
+			engine:       nil,
+			shallPass:    false,
+		},
+		{
+			name:         "fake engine struct",
+			rpcSocket:    -1,
+			containerPid: -1,
+			engine:       &fakeEngine,
+			shallPass:    false,
+		},
+		{
+			name:         "fake engine; fake rpcSocket",
+			rpcSocket:    42000,
+			containerPid: -1,
+			engine:       &fakeEngine,
+			shallPass:    false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			go doRPCthread(tt.rpcSocket, tt.containerPid, tt.engine, fatalChan)
+			fatal = <-fatalChan
+			if tt.shallPass && fatal != nil {
```

Comment:
I wrote [this](https://github.com/sylabs/singularity/wiki/End-to-end-testing#check-all-cases) with E2E in mind, but it applies the same here. Can you please read that section and do what it says there regarding handling all cases?

Reply:
Sorry, I don't see a response or a code change as suggested. Can you please follow the provided link?

Reply:
I'm still waiting on an action regarding the original request.

Reply:
good enough